src/client/connect/mod.rs

```diff
@@ -5,22 +5,18 @@
 //! - A default [`HttpConnector`](HttpConnector) that does DNS resolution and
 //!   establishes connections over TCP.
 //! - Types to build custom connectors.
-use std::convert::TryFrom;
-use std::{fmt, mem};
+use std::fmt;
 
-use bytes::{BufMut, Bytes, BytesMut};
-use ::http::{uri, Response, Uri};
+use ::http::{Response};
 
 #[cfg(feature = "tcp")] pub mod dns;
 #[cfg(feature = "tcp")] mod http;
 #[cfg(feature = "tcp")] pub use self::http::{HttpConnector, HttpInfo};
 
-/// A set of properties to describe where and how to try to connect.
-///
-/// This type is passed an argument to connectors.
-#[derive(Clone, Debug)]
-pub struct Destination {
-    pub(super) uri: Uri,
+/// Describes a type returned by a connector.
+pub trait Connection {
+    /// Return metadata describing the connection.
+    fn connected(&self) -> Connected;
```

Comment:
I feel like we could add something just like this to http-connection, then we could stay on the same trait?

Reply:
Does something else currently use `http-connection`?

Reply:
Not exactly that I know of but its something we _could_ use for this and to standardize connectors. Just thought if we want to introduce it now is the time before `0.13`. I am happy to not do it though just thinking that if we want to have external tower connectors http-connection is the best since its easy to depend on without having to bring in hyper.

Reply:
It was being used in `tower-http` but not sure what the future of that is.

Reply:
I personally would feel rushed exposing the internals of this stuff to be usable cross-crate. It can be something to plan for next time?

Reply:
Sure! Happy to punt and disucss later. I don't think this is blocking for tonic either since I manually configure the http settings both client and server side.